src/main/webapp/user_profile_script.js

```diff
@@ -12,17 +12,67 @@
 // limitations under the License.
 
-function deleteRow(row) {
-    if (confirm("Are you sure that you want to delete this already published task?")) {
-        var rowIndex = row.parentNode.parentNode.rowIndex;
-        document.getElementById("need-help").deleteRow(rowIndex);
+async function getInfo(keyString) {
+    const queryURL = "/tasks/info?key=" + keyString;
+    const request = new Request(queryURL, {method: "GET"});
+    const response = await fetch(request);
+    const info = await response.json();
+    console.log(info);
+    return info;
+}
+
+async function deleteTask(keyString) {
+    const info = await getInfo(keyString);
+    if (info.status !== "OPEN") {
+        window.alert("You can only delete an 'OPEN' task.")
+    } else {
+        if (confirm("Are you sure that you want to delete the task?")) {
+            const queryURL = "/tasks?key=" + keyString;
+            const request = new Request(queryURL, {method: "DELETE"});
+            const response = await fetch(request);
+        }
+    }
+}
+
+async function editTask(keyString) {
+    const info = await getInfo(keyString);
+    console.log(info);
+    if (info.status !== "OPEN") {
+        window.alert("You can only edit an 'OPEN' task.")
+    } else {
+        document.getElementById("edit-content-input").value = info.detail;
+        document.getElementById("edit-point-input").value = info.reward.toString();
+        const id_input = document.getElementById("task-id-input");
+        id_input.value = info.keyString;
+        id_input.readOnly = true;
+        id_input.style.display = "none"
+        document.getElementById("editTaskModal").style.display = "block";
     }
 }
 
-function completeTask(row) {
-    if (confirm("Are you sure that you have already completed the task?")) {
-        var rowIndex = row.parentNode.parentNode.rowIndex;
-        document.getElementById("offer-help").deleteRow(rowIndex);
+async function completeTask(keyString) {
+    const info = await getInfo(keyString);
+    if (info.status !== "IN PROGRESS") {
+        window.alert("You have already marked the task as complete.");
+    } else {
+        if (confirm("Are you sure that you have already completed the task?")) {
+            const queryURL = "/tasks/info?key=" + keyString + "&status=" + "COMPLETE: AWAIT VERIFICATION";
+            const request = new Request(queryURL, {method: "POST"});
+            const response = await fetch(request);
+        }
+    }
+}
+
+async function abandonTask(keyString) {
+    const info = await getInfo(keyString);
+    if (info.status !== "IN PROGRESS") {
+        window.alert("You have already marked the task as complete.");
+    } else {
+        if (confirm("Are you sure that you want to abandon the task?")) {
+            const queryURL = "/tasks/info?key=" + keyString + "&status=" + "OPEN";
+            const request = new Request(queryURL, {method: "POST"});
+            const response = await fetch(request);
```

Comment:
And here: maybe a refresh-row function would be useful.

Reply:
I force the offer help table to load again here in the new commit.